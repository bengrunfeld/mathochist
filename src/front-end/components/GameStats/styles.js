import s from 'styled-components'

const GameStatsContainer = s.div`
  padding-right: 45px !important;
`

export { GameStatsContainer }
