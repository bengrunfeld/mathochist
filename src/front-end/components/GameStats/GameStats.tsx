import React from 'react'

import GameTime from '../GameTime'
import Score from '../Score'

import { GameStatsContainer } from './styles'

const GameStats = () => (
  <GameStatsContainer className='col-6 d-flex justify-content-end'>
    <GameTime />
    <Score />
  </GameStatsContainer>
)

export default GameStats
