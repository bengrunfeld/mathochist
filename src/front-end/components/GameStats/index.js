import GameStats from './GameStats'

export default GameStats
