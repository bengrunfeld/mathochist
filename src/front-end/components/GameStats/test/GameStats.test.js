import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

import GameStats from '../GameStats'

describe('<GameStats />', () => {
  test('matches the snapshot', () => {
    const tree = mount(<GameStats />)
    expect(tree).toMatchSnapshot()
  })
})
