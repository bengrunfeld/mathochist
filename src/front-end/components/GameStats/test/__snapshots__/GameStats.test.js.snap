// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameStats /> matches the snapshot 1`] = `
<GameStats>
  <div
    className="col-6"
  >
    <GameTime>
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "lastClassName": "dmRLcz",
                "rules": Array [
                  "
  display: inline-block;
  margin-right: 35px;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="sc-htpNat dmRLcz"
          >
            <StatsBox
              data="0"
              statsType="time"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bdVaJa",
                        "isStatic": true,
                        "lastClassName": "iKOMUo",
                        "rules": Array [
                          "
  display: inline-block;
  background: #b7b7b7;
  padding: 6px;
  border: 2px solid black;
  min-width: 70px;
  border-radius: 5px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bdVaJa",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-bdVaJa iKOMUo"
                  >
                    <styled.p
                      type="time"
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bwzfXH",
                              "isStatic": false,
                              "lastClassName": "bREitx",
                              "rules": Array [
                                "
  font-size: 30px;
  font-weight: bold;
  font-family: Nunito;
  margin: 0;
  padding: 0;
  text-align: center;

  color: ",
                                [Function],
                                ";
",
                              ],
                            },
                            "displayName": "styled.p",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-bwzfXH",
                            "target": "p",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        type="time"
                      >
                        <p
                          className="sc-bwzfXH jlhPjW"
                          type="time"
                        >
                          0
                        </p>
                      </StyledComponent>
                    </styled.p>
                  </div>
                </StyledComponent>
              </styled.div>
            </StatsBox>
            <styled.h4>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bxivhb",
                      "isStatic": true,
                      "lastClassName": "iDhuiM",
                      "rules": Array [
                        "
  text-align: center;
  font-size: 22px;
  margin-top: 5px;
",
                      ],
                    },
                    "displayName": "styled.h4",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bxivhb",
                    "target": "h4",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <h4
                  className="sc-bxivhb iDhuiM"
                >
                  Time
                </h4>
              </StyledComponent>
            </styled.h4>
          </div>
        </StyledComponent>
      </styled.div>
    </GameTime>
    <Score
      playerScore="0"
    >
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "lastClassName": "kmbVwf",
                "rules": Array [
                  "
  display: inline-block;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="sc-ifAKCX kmbVwf"
          >
            <StatsBox
              data="0"
              statsType="score"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bdVaJa",
                        "isStatic": true,
                        "lastClassName": "iKOMUo",
                        "rules": Array [
                          "
  display: inline-block;
  background: #b7b7b7;
  padding: 6px;
  border: 2px solid black;
  min-width: 70px;
  border-radius: 5px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bdVaJa",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-bdVaJa iKOMUo"
                  >
                    <styled.p
                      type="score"
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bwzfXH",
                              "isStatic": false,
                              "lastClassName": "bREitx",
                              "rules": Array [
                                "
  font-size: 30px;
  font-weight: bold;
  font-family: Nunito;
  margin: 0;
  padding: 0;
  text-align: center;

  color: ",
                                [Function],
                                ";
",
                              ],
                            },
                            "displayName": "styled.p",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-bwzfXH",
                            "target": "p",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        type="score"
                      >
                        <p
                          className="sc-bwzfXH bREitx"
                          type="score"
                        >
                          0
                        </p>
                      </StyledComponent>
                    </styled.p>
                  </div>
                </StyledComponent>
              </styled.div>
            </StatsBox>
            <styled.h4>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-EHOje",
                      "isStatic": true,
                      "lastClassName": "bLIEbZ",
                      "rules": Array [
                        "
  text-align: center;
  font-size: 22px;
  margin-top: 5px;
",
                      ],
                    },
                    "displayName": "styled.h4",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-EHOje",
                    "target": "h4",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <h4
                  className="sc-EHOje bLIEbZ"
                >
                  Score
                </h4>
              </StyledComponent>
            </styled.h4>
          </div>
        </StyledComponent>
      </styled.div>
    </Score>
  </div>
</GameStats>
`;
